src/world/map.c

```diff
@@ -1709,10 +1970,163 @@ static money32 map_set_land_height(int flags, int x, int y, int height, int styl
 			return MONEY32_UNDEFINED;
 		}
 	}
+	
+	/*	Continuing decompilation - work in progress
+	 *	Known bugs (introduced by disassembly):
+	 *		none! Still requires testing, though.
+	 *	Original bugs:
+	 *		Lowering the land tile underneath a wall removes the wall.
+	 *		Land can be raised over flat scenery items (gardens, for example) without removing them.
+	 *		Removal price of walls is not added.
+	 *		Removal price of flat scenery is not counted when raising the center of a tile.
+	 */
+	
+	int eax = x, ebx = flags, ecx = y, edi = selectionType << 5, ebp = 0;
+	uint32 edx = (style << 8) | height;
+	//dl: height
+	//dh: style
+	uint32 esi = 0;
+	
+	//Uncomment to use vanilla code
+	//RCT2_CALLFUNC_X(0x006639FE, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+	
+	RCT2_GLOBAL(0x9E2E18, money32) = MONEY(0, 0);
+	if(ebx&GAME_COMMAND_FLAG_APPLY)
+	{
+		footpath_remove_litter(x, y, map_element_height(x, y));
+		map_remove_walls_at(x, y, height*8-16, height*8+32);
+	}
+	RCT2_GLOBAL(0x9E2E18, money32) += MONEY(20, 0);
+	
+	rct_map_element *mapElement = map_get_first_element_at(x/32, y/32); //mapElement = esi
+	
+	int zHigh = height+4; //zHigh = dh
+	
+//loc_663A54:
+	do{
+		int elementType = map_element_get_type(mapElement); //elementType = eax
+		rct_scenery_entry *sceneryEntry; //eax
+		
+		if(elementType!=MAP_ELEMENT_TYPE_SCENERY || height>mapElement->clearance_height || zHigh<(mapElement)->base_height)
```

Comment:
can you split this up into multiple `if`s. I've changed all of map.c to be in the same format and this would be inconsistent. I'd prefer it if you didn't have a separate var for elementType and just used it in the if since its only used ones.